src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs

```diff
@@ -990,8 +990,11 @@ private bool IsContractCheckMethod(IMethodSymbol method)
             return false;
         }
 
-        private bool IsDebugAssertMethod(IMethodSymbol method)
-            => Equals(method, DebugAssertMethod);
+        private bool IsDebugAssertMethod(IMethodSymbol method) =>
+            DebugAssertMethod != null &&
+            method.ContainingSymbol.Equals(DebugAssertMethod.ContainingSymbol, SymbolEqualityComparer.Default) &&
+            method.Name == DebugAssertMethod.Name &&
+            method.ReturnType == DebugAssertMethod.ReturnType;
```

Comment:
SymbolEqualityComparer? I'm surprised `==` actually works.

Reply:
Did not undertsood, for `SymbolEqualityComparer` I used `SymbolEqualityComparer.Equals` not `==`

Reply:
@buyaa-n I usee you used SymbolEqualityComparer for comparing `ContainingSymbol`s, but not for `ReturnType`s.

> method.ReturnType == DebugAssertMethod.ReturnType;